configs/common/Simulation.py

```diff
@@ -52,9 +56,11 @@
 addToPath("../common")
 
 
-def getCPUClass(cpu_type):
+def getCPUClass(cpu_type: str, isa: str):
     """Returns the required cpu class and the mode of operation."""
-    cls = ObjectList.cpu_list.get(cpu_type)
+    cls = SimpleCore.cpu_class_factory(
+        cpu_type=get_cpu_type_from_str(cpu_type), isa=get_isa_from_str(isa)
+    )
```

Comment:
Note: I'm aware that this isn't all that elegant, I should really move this function out of `SimpleCore` in the stdlib, but it works for now and didn't want to take on that right now.

It's probably worth providing some more factories to allow people to select and configure cores like this.

Reply:
Hi Bobby, it doesn't look unelegant to me, the only issue is that the ObjectList contains a wider set of cpus (from the configs subdirectory) like HPI as an example...

Couldn't we still rely on the ObjectList module and let the user choose the ISA implicitly via the CPU type? I haven't checked but I presume that when you build with ALL you will have something like

ArmAtomicCPU, ArmTmingCPU, [...], RiscvAtomicCPU, RiscvTimingCPU, [...]

Let me know if I am missing something

Reply:
I've added code here to utilize ObjectList (which is kind of doing the same thing. My only worry here is we're hard-coding our naming scheme here. Like our CPU implementations having the name `{{ISA in Camel Case}}{{CPU Type}}`. If you're ok with this though, i'm fine with it too.

Reply:
IMO, this is OK for the "legacy" configuration scripts. I *really* want to see us focus our effort solely on the standard library because it's going to be much easier to maintain going forward. So, I think it's OK to use relatively brittle things like assuming a particular naming scheme in these legacy scripts to let them limp along a little longer.